File: clients/py/src/features/wrap_type_map.py
from pathlib import Path
from typing import Any, Dict

from polywrap import (
    FsUriResolver,
    PolywrapClient,
    PolywrapClientConfigBuilder,
    SimpleFileReader,
    Uri,
)
from pydantic import BaseModel


class TestCaseInput(BaseModel):
    map: Dict[str, int]


def run_test_case(input: Any) -> None:
    input_obj = TestCaseInput.parse_obj(input)

    root = Path(__file__).parent.parent.parent.parent.parent / "wraps"
    uri = Uri.from_str(f"fs/{root}/map-type/implementations/as")

    config = (
        PolywrapClientConfigBuilder()
        .add_resolver(FsUriResolver(SimpleFileReader()))
        .build()
    )

    client = PolywrapClient(config)

    print("Invoking returnMap")

    response = client.invoke(
        uri=uri,
        method="returnMap",
        args={"map": input_obj.map},
    )

    if not response:
        raise ValueError(f"Error: {response}")

    returned_map = response

    for key in input_obj.map:
        print(f"key '{key}' = {returned_map.get(key)}")

    print("Success!")

File: clients/js/src/features/wrap_type_map.ts
import { Input } from "../input";

import { PolywrapClientConfigBuilder, PolywrapClient } from "@polywrap/client-js";
import path from "path";

export async function runTestCase(input: unknown): Promise<void> {
  const inputObj = Input.expectObject<{
    map: unknown;
  }>(input);
  const map = Input.expectObject<Record<string, unknown>>(
    inputObj.map
  );

  const root = path.join(__dirname, "../../../../wraps");
  const uri = `fs/${root}/map-type/implementations/as`;

  const config = new PolywrapClientConfigBuilder()
    .addDefaults()
    .build();
  const client = new PolywrapClient(config);

  const mapClass = new Map<string, number>();

  for (const entry of Object.entries(map)) {
    mapClass.set(entry[0], entry[1] as number);
  }

  console.log("Invoking returnMap");

  const returnMapResponse1 = await client.invoke<Map<string, number>>({
    uri,
    method: "returnMap",
    args: {
      map: mapClass,
    },
  });

  if (!returnMapResponse1.ok) throw returnMapResponse1.error;

  const returnedMap = returnMapResponse1.value;

  for (const entry of Object.entries(map)) {
    console.log(`key '${entry[0]}' = ${returnedMap.get(entry[0])}`);
  }

  console.log("Success!");
}
